### PR TITLE
Added `to-absolute` method to length.

### PR DESCRIPTION
**Pull request by Dherse**
Adds the `.to-absolute(styles)` method to `length` to allow retrieving a length as absolute (without the `em` part) when in the right context.

This changes fixes packages like tablex (https://github.com/PgBiel/typst-tablex) that use the following code:
```typ
#let measure-pt(len, styles) = {
    let measured-pt = measure(box(width: len), styles).width

    if measured-pt > 0pt {
        return measured-pt
    }

    measured-pt = -measure(box(width: -len), styles).width
    return measured-pt
}
```

### CODE REVIEW

File: crates/typst/src/layout/length.rs [hunk 1]
```diff
@@ -132,6 +132,32 @@ impl Length {
...
+            abs: self.abs + self.em.resolve(styles),
+            em: Em::zero(),
+        })
+    }
+
+    /// Returns whether this length is absolute (that is, has no `em` units).
+    #[func]
+    pub fn is_absolute(&self) -> bool {
+        self.em.is_zero()
+    }
```
Comment (laurmaedje) [inline crates/typst/src/layout/length.rs:160]: I don't think this method should be added. It's trivial to write in Typst, too, by accessing the `em` field.

**Review by laurmaedje — COMMENTED** [review]
I'm okay with adding this, but think we should discuss the name a bit. `resolve` is quite generic. Maybe we can find a name that describes more concretely what is happening.

**Comment by istudyatuni** [comment]
`measure`, but as a method?

**Comment by Dherse** [comment]
> `measure`, but as a method?

We discussed this on Discord quickly with @laurmaedje and agreed that `to-absolute` was a better name since `resolve` is too generic and `measure` carries a different meaning.

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**